ci/jenkins/jobs/projects-cloud.yaml

```diff
@@ -741,7 +751,64 @@
           - email:
               notify-every-unstable-build: true
               recipients: projectantrea-dev@googlegroups.com
-          wrappers: []
+          wrappers:
+          - credentials-binding:
+            - text:
+                credential-id: WORKFORCE_POOL # Jenkins secret that stores the cloud resource pool id
+                variable: WORKFORCE_POOL
+      - 'cloud-{name}-{test_name}-credential':
+          test_name: gke
+          node: antrea-credential
+          description: 'This is the cloud credential job for antrea gke tests.'
+          builders:
+          - shell: |-
+              #!/bin/bash
+              # Prevent exposing credentials in the console output by adding set +x
+              # This is to avoid other developers removing this set+x by accident
+              set +x
+              token_file="/var/lib/jenkins/ci_properties.txt"
+              if [ -e "$token_file" ]; then
+                  modification_timestamp=$(stat -c %Y "$token_file")
+                  current_timestamp=$(date +%s)
+                  time_difference=$((current_timestamp - modification_timestamp))
+              else
+                  time_difference=36001
+              fi
+              # The credentials are valid for 12 hours, to allow sufficient time for job execution, the reuse threshold is set to 10 hours.
```

Comment:
How can we run gke test for specific commit in the future? I think the PR changes it to not retrieve credential on-demand, while there is no way to specify repo and revision in antrea-credential job and it shouldn't need to have that.
I wonder why it changes to persist the credential instead of getting it on demand. Even it's the more recommended way, should we just have an asynchronous job to refresh the credential and let the test job run as before?

Reply:
> How can we run gke test for specific commit in the future? I think the PR changes it to not retrieve credential on-demand, while there is no way to specify repo and revision in antrea-credential job and it shouldn't need to have that.

The credential job's purpose is to refresh credentials periodically (every 12 hours). For the previous GKE job, we can still trigger it with specified repo and revision, as the config file for 'gcloud login' has been updated by the credential job.

> I wonder why it changes to persist the credential instead of getting it on demand. Even it's the more recommended way, should we just have an asynchronous job to refresh the credential and let the test job run as before?

Since the credential node and the cloud node are on different networks, we have to use a Jenkins post-action to transmit the credentials securely.

Reply:
But if the credential is refreshed every 12 hours, and it triggers a test via post-action every time, the test actually runs 2 times per day, instead of 1 time per two days?

Reply:
GKE job won't be triggered automatically; it can only be triggered manually after this code change.

Reply:
Then I'm confused by the triggers "H H/2 * * *" and the publishers of the new job. Doesn't it mean it will refresh the credential every 12 hours and trigger a GKE job after it?

And how do we manually run GKE job, via the GKE job itself or the credential job?

cc @antoninbas for awareness if GKE job really won't run automatically.

Reply:
> Then I'm confused by the triggers "H H/2 * * *" and the publishers of the new job. Doesn't it mean it will refresh the credential every 12 hours and trigger a GKE job after it?

I have updated the cron expression to original value(H H */2 * *) , the cloud job will be triggered by the credential job every two days.

> And how do we manually run GKE job, via the GKE job itself or the credential job?

You can still manually trigger the GKE job using the provided token ID. For the credential job, running it manually will trigger the cloud test based on the main branch.

Reply:
yes, we should keep running the job periodically, and it seems to be the case now

> I have updated the cron expression to original value(H H */2 * *) , the cloud job will be triggered by the credential job every two days.

It seems unnecessary to tie the 2 jobs, they could just execute periodically independently from each other, and they don't even need to execute with the same frequency.

Reply:
> It seems unnecessary to tie the 2 jobs, they could just execute periodically independently from each other, and they don't even need to execute with the same frequency.

The reason we tie these two jobs together is that the cloud testbed and credential testbed are on different networks (the credential script can only work in the lab network), and the credential cannot be transmitted by the SCP command(both cloud and credential node have no public IP). We have to use Jenkins post-action to safely deliver the credential to the cloud node. If you prefer the credentials on the cloud node to always be up-to-date, I can create a new periodic job on the cloud node as the post-action of the credential job to update the credentials on the cloud node every 6 hours. Then we can maintain the original behavior of the GKE job. What do you think? @antoninbas

Reply:
That sounds ok to me if it's not too much work. The most important things are that we can run the test manually and that the test also runs periodically.

Reply:
> That sounds ok to me if it's not too much work. The most important things are that we can run the test manually and that the test also runs periodically.

Sure, updated code, PTAL.